llvm-projects/iree-compiler-backend/build_tools/smoketest.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from iree.compiler_backend import ir
```

Comment:
curious: why "backend"? if anything, this is the frontend - backend is used in IREE to denote the actual backend compilers we target, and I'm worried about aliasing that here
could this just be `iree.compiler`?

Reply:
Oof, yeah. We already have an `iree.compiler` as a more user-facing API (which also includes access to importers, etc). I do envision this more as an "integrator's" toolkit. Front and back is relative, and I was thinking of this in terms of a backend to that frontend. But yeah... bad choice. Maybe `iree.compiler_frontend`?

Reply:
What about `iree.mlir` or `iree.compiler.mlir`?

* `iree.compiler` for the high level "I just want to convert my model to include it in my app" API
* `iree.mlir` or `iree.compiler.mlir` for the lower level "I want to integrate with another framework through MLIR" or "I want to access the lower level MLIR ops, passes, etc." API

Reply:
Chatted with Ben: I think we're just going with `iree.compiler` and plan for this to incorporate the existing package. There really isn't a reason to bifurcate (and a lot of reason to not, since within a package, you get free access to native resources, etc).

This doesn't become a problem until we decide to deploy these artifacts, and we're a ways from that (beyond experimentation). When we do, we will need to deploy this and delete the current iree.compiler. This is good news anyway since that one is built wrong (and would need to be redone at some point) and this one is built right.